File: app/src/main/java/com/shouzhan/design/ui/home/DraggerActivity.java
package com.shouzhan.design.ui.home;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.View;
import com.fshows.android.stark.utils.FsLogUtil;
import com.shouzhan.design.R;
import com.shouzhan.design.base.BaseActivity;
import com.shouzhan.design.databinding.ActivityDraggerBinding;
import com.shouzhan.design.model.javabean.Student;

import javax.inject.Inject;


/**
 * @author danbin
 * @version DraggerActivity.java, v 0.1 2019-02-27 上午12:11 danbin
 */
public class DraggerActivity extends BaseActivity<ActivityDraggerBinding> {

    private static final String TAG = DraggerActivity.class.getSimpleName();

    @Inject
    Student mStudent;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        getData();
    }

    @Override
    public void onClick(View view) {

    }

    @Override
    public int getLayoutId() {
        return R.layout.activity_dragger;
    }

    @Override
    public void getData() {
        FsLogUtil.error(TAG, mStudent.name);
    }

}

File: app/src/main/java/com/shouzhan/design/ui/home/DraggerActivity.kt
package com.shouzhan.design.ui.home

import android.app.Activity
import android.arch.lifecycle.ViewModelProvider
import android.os.Bundle
import android.view.View
import com.shouzhan.design.R
import com.shouzhan.design.base.BaseActivity
import com.shouzhan.design.databinding.ActivityMainBinding
import com.shouzhan.design.ui.home.viewmodel.DraggerViewModel
import dagger.android.AndroidInjector
import dagger.android.DispatchingAndroidInjector
import dagger.android.HasActivityInjector
import kotlinx.android.synthetic.main.activity_main.*
import javax.inject.Inject


/**
 * @author danbin
 * @version DraggerActivity.java, v 0.1 2019-02-27 上午12:11 danbin
 */
class DraggerActivity : BaseActivity<ActivityMainBinding>(), HasActivityInjector {

    override fun onClick(view: View?) {
    }

    @Inject
    internal var dispatchingAndroidInjector: DispatchingAndroidInjector<Activity>? = null

    override fun activityInjector(): AndroidInjector<Activity>? {
        return dispatchingAndroidInjector
    }

    @Inject
    internal lateinit var viewModelFactory: ViewModelProvider.AndroidViewModelFactory

    private val homeViewModel by lazy(LazyThreadSafetyMode.NONE) {
        viewModelFactory.create(DraggerViewModel::class.java)
    }

    override fun getLayoutId(): Int {
        return R.layout.activity_dragger
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        getData()
    }

    override fun initView() {
        super.initView()
        select_btn.setOnClickListener(this)
    }

    override fun getData() {
        homeViewModel.requestHomeData(1)
    }

}
